tools/env-loader/pkg/writers/gha-mask.go

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright 2024 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package writers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/shared-workflows/tools/env-loader/pkg/values"
+	"github.com/gravitational/trace"
+)
+
+const ghaMaskPrefix = "::add-mask::"
+
+// Outputs secret values prefixed with `::add-mask::`, one per line.
+// Per GHA docs, this will prevent the values from being logged.
+// For details, see
+// https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#masking-a-value-in-a-log
+type GHAMaskWriter struct{}
+
+// Create a new GHA mask writer
+func NewGHAMaskWriter() *GHAMaskWriter {
+	return &GHAMaskWriter{}
+}
+
+func (ew *GHAMaskWriter) FormatEnvironmentValues(values map[string]values.Value) (string, error) {
+	renderedValues := make([]string, 0, len(values))
+	for key, value := range values {
+		if key == "" {
```

Comment:
This seems strange that the key emptiness check is inside this specific writer. Is there a way to validate the key validity when loading the value rather than in each specific writer?

Reply:
While this conceptually makes, I think it would add some unnecessary complexity. Here's what I've done instead:
* Moved all the logic from the `if` statement to the `String()` function
* Added tests for _all_ writers to verify that they contain logic that covers this case

All writers still need to handle the case of `key == ""`, but they can do it in their own way, without additional complexity.